pydantic/datetime_parse.py

```diff
@@ -226,7 +226,14 @@ def parse_duration(value: StrBytesIntFloat) -> timedelta:
         value = str(value)
     elif isinstance(value, bytes):
         value = value.decode()
-
+    
+    if isinstance(value, (str)):
```

Comment:
Instead of adding this into the logic for all cases, I think you can match all the cases described in your tests with another regex, which can be appended to the current regexes checked and therefore have no effect on performance if it's not called.

Reply:
Thanks for you feedback. 

The regex in the commit  https://github.com/samuelcolvin/pydantic/pull/3346/commits/88ffae7f8cf14bd33f3a22f57cc302c2a3021aaf  is good for you?

Reply:
please review